ikasaneip/component/endpoint/mongo4-endpoint/src/main/java/org/ikasan/component/endpoint/mongo/MongoClientFactory.java

```diff
@@ -0,0 +1,82 @@
+package org.ikasan.component.endpoint.mongo;
```

Comment:
Is this a good idea to reuse same package name? 

Could you have module where both mongo3 and mongo4 coexist ?